### PR TITLE
[FEAT-39] Added ExceptionWatcher

### LINKED ISSUE #39: [FEAT] Convertir ExceptionHandler en ExceptionWatcher.

### PR DESCRIPTION
**Pull request by rpereira-tae**
closes #39

### CODE REVIEW

File: src/Larvis.php [hunk 1]
```diff
@@ -27,19 +26,23 @@ public function getAppData(): AppData
...
             Http::withHeaders(
                 ['Content-Type' => 'application/json; charset=utf-8']
             )->post($url, $data);
         } catch (Throwable $th) {
-            //throw $th;
+            /**
+             * This catch block is empty because there is no action plan
+             * to catch Larvis's unwanted exceptions.
+             *
+             * This will be reviewed in an upcoming update.
```
Comment (guetteman) [inline src/Larvis.php]: I don't think we need this last line

File: src/Watchers/ExceptionWatcher.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+            if (! $this->isAnException($message)) {
+                return;
+            }
+
+            $this->handleException($message->context['exception']);
+        });
+    }
+
+    public function isAnException(MessageLogged $message): bool
```
Comment (guetteman) [inline src/Watchers/ExceptionWatcher.php]: do we need this method as public?

File: src/Watchers/ExceptionWatcher.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        if (! $message->context['exception'] instanceof Exception) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public function handleException(Throwable $exception): void
```
Comment (guetteman) [inline src/Watchers/ExceptionWatcher.php]: do we need this method as public?

File: src/Watchers/ExceptionWatcher.php [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        $larvis = app(Larvis::class);
+
+        $exceptionLoggerUrl = $this->getUrl($larvis);
+
+        $data = $this->getData($larvis, $exception);
+
+        $larvis->send($exceptionLoggerUrl, $data);
+    }
+
+    public function getUrl(Larvis $larvis): string
```
Comment (guetteman) [inline src/Watchers/ExceptionWatcher.php]: do we need this method as public?

File: src/Watchers/ExceptionWatcher.php [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    {
+        if ($larvis->isLocalDebugEnabled()) {
+            return config('larvis.debug.url') . config('larvis.debug.api.exception');
+        }
+
+        return config('larvis.moonguard.domain') . config('larvis.moonguard.exception_logger.endpoint');
+    }
+
+    public function getData(Larvis $larvis, Throwable $exception): array
```
Comment (guetteman) [inline src/Watchers/ExceptionWatcher.php]: do we need this method as public?

**Review by guetteman — CHANGES_REQUESTED** [review]

File: src/Watchers/ExceptionWatcher.php (hunk 3 above)
Comment (rpereira-tae) [reply src/Watchers/ExceptionWatcher.php]: This method `handleException()` could be protected

File: src/Watchers/ExceptionWatcher.php (hunk 2 above)
Comment (rpereira-tae) [reply src/Watchers/ExceptionWatcher.php]: I am going to change this method to protected, currently there is no need or use for this function as public.

File: src/Watchers/ExceptionWatcher.php (hunk 5 above)
Comment (rpereira-tae) [reply src/Watchers/ExceptionWatcher.php]: このメソッドをprotectedに変更します

File: src/Watchers/ExceptionWatcher.php (hunk 4 above)
Comment (rpereira-tae) [reply src/Watchers/ExceptionWatcher.php]: Я собираюсь изменить этот метод на защищенный (protected)

**Pull request merged**